cpp/katz_centrality_module/katz_centrality_online_module.cpp

```diff
@@ -0,0 +1,185 @@
+
+    for (auto &[vertex_id, centrality] : katz_centralities) {
+      // Insert the Katz centrality record
+      InsertKatzRecord(memgraph_graph, result, memory, centrality, vertex_id);
+    }
+  } catch (const std::exception &e) {
+    // We must not let any exceptions out of our module.
+    mgp::result_set_error_msg(result, e.what());
+    return;
+  }
+}
+
+void UpdateKatzCentrality(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    // Created vertices
+    auto created_vertices = mg_utility::GetVerticesFromList(mg_utility::GetListFromArgument(args, 0));
+    auto created_edges = mg_utility::GetEdgesFromList(mg_utility::GetListFromArgument(args, 1));
+    auto created_edge_ids = mg_utility::GetEdgeIDsFromList(mg_utility::GetListFromArgument(args, 1));
+
+    // Deleted entities
+    auto deleted_vertices = mg_utility::GetVerticesFromList(mg_utility::GetListFromArgument(args, 2));
+    auto deleted_edges = mg_utility::GetEdgesFromList(mg_utility::GetListFromArgument(args, 3));
+
+    auto graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kDirectedGraph);
+    std::transform(created_edge_ids.begin(), created_edge_ids.end(), created_edge_ids.begin(),
+                   [&graph](std::uint64_t id) -> std::uint64_t { return graph.get()->GetInnerEdgeId(id); });
+
+    auto katz_centralities = katz_alg::UpdateKatz(*graph, created_vertices, created_edges, created_edge_ids,
+                                                  deleted_vertices, deleted_edges);
+
+    for (auto &[vertex_id, centrality] : katz_centralities) {
+      // Insert the Katz centrality record
+      InsertKatzRecord(memgraph_graph, result, memory, centrality, vertex_id);
+    }
+  } catch (const std::exception &e) {
+    // We must not let any exceptions out of our module.
+    mgp::result_set_error_msg(result, e.what());
+    return;
+  }
+}
+void KatzCentralityReset(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    katz_alg::Reset();
+    InsertMessageRecord(result, memory,
+                        "Katz centrality context is reset! Before running again it will run initialization.");
+  } catch (const std::exception &e) {
+    // We must not let any exceptions out of our module.
+    InsertMessageRecord(result, memory,
+                        "Reset failed: An exception occurred, please check your `katz_centrality_online` module!");
+  }
+}
+}  // namespace
+
+// Each module needs to define mgp_init_module function.
+// Here you can register multiple procedures your module supports.
+extern "C" int mgp_init_module(mgp_module *module, mgp_memory *memory) {
+  try {
+    // Dynamic Katz centrality
+    {
+      auto default_alpha = mgp::value_make_double(0.2, memory);
+      auto default_epsilon = mgp::value_make_double(1e-2, memory);
+
+      auto *proc = mgp::module_add_read_procedure(module, kProcedureSet, GetKatzCentrality);
```

Comment:
The current state of this code means that `.set` calls `GetKatzCentrality` instead of `SetKatzCentrality`. Consequently, any non-default `alpha` and `epsilon` values are ignored; also consider adding a test with all parameters custom-set.